js/PixelPainter.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Creates a grid of elements.
+ * @param {number} rows – The number of rows in the grid.
+ * @param {number} columns – The number of columns in the grid.
+ * @param {object} attributes - The attributes to apply to each grid element
+ * @returns {HTMLElement} – The DOM element containing the grid of divs
+ */
+function createGrid (rows, columns, attributes) {
+  // row checks
+  if(rows < 0) {
+    throw new Error("Rows must be a positive number.");
+  }
+  else if(rows === undefined) {
+    throw new Error("Rows must be defined.");
+  }
+  else if(typeof rows === 'object') {
+    throw new Error("Rows cannot be an empty Object.");
+  }
+
+  // column checks
+  if(columns === undefined && attributes === undefined) {
+    columns = 1;
+    attributes = {};
+  }
+  else if(columns === undefined) {
+    columns = 1;
+    attributes = {};
+  }
+  else if(typeof columns !== 'number') {
+    attributes = columns;
+    columns = rows;
+  }
+  else if(attributes === undefined) {
+    attributes = {};
+  }
+
+  var grid = document.createElement("div");
+  var row = document.createElement("div");
+  var column = document.createElement("div");
+
+  var rowCount = 0;
+  var columnCount = 0;
+  var att;
+  var attCount = 0;
+
+  // row.className = "row";
+  // column.className = "column";
+
+  while(rowCount < rows) {
+    while(attCount < (Object.keys(attributes).length)) {
```

Comment:
This might be a little overkill, attributes only really need to be applied to the column elements. It might get kind of hard to manage if the things like mouse events and styles  that  are passed in get applied to both the row *and* the column that you're clicking on.